### ISSUE TITLE
[Bug][SentMails] Do not appear in sent mails of mail document

### ISSUE THREAD

**Issue by fashxp** [pimcore/pimcore#10354]
### Expected behavior

do appear in sent mails list of mail document

![image](https://user-images.githubusercontent.com/8792145/133975806-cb43a654-1b2a-429c-8624-f320882b45e9.png)

### Actual behavior

only appear in global list

### Steps to reproduce

send document mail... e.g. for demo
```
        $params = [
            'customerId' => 1018,
            'tokenLink' => 'fooo'
        ];

        $mail = new \Pimcore\Mail(['document' => '/en/mails/password-recovery', 'params' => $params]);
        $mail->addTo('YOUR_MAIL_ADDRESS');
        $mail->send();
```

**Comment by dvesh3** [comment]
Regression of https://github.com/pimcore/pimcore/pull/9838

**Comment by brusch** [comment]
Fixed by #10378
